NOnion/Directory/TorDirectory.fs

```diff
@@ -385,7 +385,9 @@ type TorDirectory =
                          |> HiddenServicesCipher.SHA3256)
                         |> Some
                     with
-                    | _ -> None
+                    | :? System.Collections.Generic.KeyNotFoundException ->
```

Comment:
@aarani what piece of code is using SCG here?

Reply:
self.GetDescriptorByIdentity uses F#'s Map.[index] which can throw KeyNotFoundException

Reply:
then the correct fix for this is not catch KeyNotFoundException, but use tryFind instead, like this FSharpLint recommends: https://github.com/fsprojects/FSharpLint/commit/e4dcdaed07ca4c4145e8aa71cac1d02014df8a3d

Reply:
Removed in https://github.com/nblockchain/NOnion/pull/30/commits/242350dec2d54ff9d4382c9ec9c21602d1659c83.